hugr-core/src/hugr/views/sibling_subgraph.rs

```diff
@@ -217,7 +222,20 @@ impl<N: HugrNode> SiblingSubgraph<N> {
         nodes: impl Into<Vec<N>>,
         hugr: &impl HugrView<Node = N>,
     ) -> Result<Self, InvalidSubgraph<N>> {
-        let checker = TopoConvexChecker::new(hugr);
+        let nodes = nodes.into();
+        let Some(node) = nodes.first() else {
+            return Err(InvalidSubgraph::EmptySubgraph);
+        };
+        let parent = hugr
+            .get_parent(*node)
+            .ok_or(InvalidSubgraph::NoSharedParent {
```

Comment:
Could you add an enum variant e.g. InvalidSubgraph::OrphanNode? Currently, the error message you'd get would be confusing (also useful in pick_parent)